src/test/java/com/firebolt/jdbc/connection/FireboltConnectionTest.java

```diff
@@ -351,28 +354,62 @@ void shouldThrowExceptionWhenCannotUnwrap() throws SQLException {
 	@Test
 	void shouldGetDatabaseWhenGettingCatalog() throws SQLException {
 		connectionProperties.put("database", "db");
-		try (Connection connection = new FireboltConnection(URL, connectionProperties, fireboltAuthenticationService,
-				fireboltEngineService, fireboltStatementService)) {
+		try (Connection connection = createConnection(URL, connectionProperties)) {
 			assertEquals("db", connection.getCatalog());
 		}
 	}
 
 	@Test
 	void shouldGetNoneTransactionIsolation() throws SQLException {
 		connectionProperties.put("database", "db");
-		try (Connection connection = new FireboltConnection(URL, connectionProperties, fireboltAuthenticationService,
-				fireboltEngineService, fireboltStatementService)) {
+		try (Connection connection = createConnection(URL, connectionProperties)) {
 			assertEquals(Connection.TRANSACTION_NONE, connection.getTransactionIsolation());
 		}
 	}
 
 	@Test
 	void shouldThrowExceptionWhenPreparingStatementWIthInvalidResultSetType() throws SQLException {
 		connectionProperties.put("database", "db");
-		try (Connection connection = new FireboltConnection(URL, connectionProperties, fireboltAuthenticationService,
-				fireboltEngineService, fireboltStatementService)) {
+		try (Connection connection = createConnection(URL, connectionProperties)) {
 			assertThrows(SQLFeatureNotSupportedException.class,
 					() -> connection.prepareStatement("any", TYPE_SCROLL_INSENSITIVE, 0));
 		}
 	}
+
+	@Test
+	void shouldGetDefaultEngineWhenEngineIsNotProvided() throws SQLException {
+		connectionProperties.put("engine", "");
+		when(fireboltEngineService.getEngine(any(), any())).thenReturn(Engine.builder().name("default_engine").endpoint("http://my-endpoint").build());
+		try (FireboltConnection  connection = createConnection(URL, connectionProperties)) {
+			verify(fireboltEngineService).getEngine("", "db");
+			assertEquals("default_engine", connection.getSessionProperties().getEngine());
+			assertEquals("http://my-endpoint", connection.getSessionProperties().getHost());
+		}
+	}
+
+	@Test
+	void shouldGetEngineUrlWhenEngineIsProvided() throws SQLException {
+		connectionProperties.put("engine", "engine");
+		when(fireboltEngineService.getEngine(any(), any())).thenReturn(Engine.builder().endpoint("http://my_endpoint").build());
+		try (FireboltConnection connection = createConnection(URL, connectionProperties)) {
+			verify(fireboltEngineService).getEngine("engine", "db");
+			assertEquals("http://my_endpoint", connection.getSessionProperties().getHost());
+		}
+	}
+
+	@Test
+	void shouldNotGetEngineUrlOrDefaultEngineUrlWhenUsingSystemEngine() throws SQLException {
+		connectionProperties.put("engine", "system");
```

Comment:
engine=system is no longer a valid parameter. The new workflow is if no engine name is specified the driver connects to a system engine.

Reply:
Let's disucuss this an fix the test together. I am not sure I understand what should be done here. Anyway this is an old test; I just improved the code style.

Reply:
Done. No more "system".